### PR TITLE
Added button to convert an operand or result to binary notation

### PR DESCRIPTION
**Pull request by mob-jen**

### CODE REVIEW

File: Calculator/Calculator/Models/MCACalculator.m [hunk 1]
```diff
@@ -76,6 +76,17 @@ - (NSDecimalNumber *)evaluateExpressionFromHistory
     return [self simplifyOperators:[self.operators mutableCopy] forOperands:[self.operands mutableCopy]];
 }
 
+- (NSString *)convertOperandToBinary:(NSDecimalNumber *)operand {
+    
+    NSString *binaryString = @"" ;
```
Comment (nielubowicz) [inline Calculator/Calculator/Models/MCACalculator.m]: Please remove blank space before semi-colon.

File: Calculator/Calculator/Models/MCACalculator.m [hunk 2]
```diff
@@ -76,6 +76,17 @@ - (NSDecimalNumber *)evaluateExpressionFromHistory
...
+    NSUInteger x = operand.integerValue ;
+    do {
+        binaryString = [[NSString stringWithFormat: @"%lu", x&1] stringByAppendingString:binaryString];
+    } while (x >>= 1);
```
Comment (nielubowicz) [inline Calculator/Calculator/Models/MCACalculator.m]: Can you explain how this loop works? I'm having trouble wrapping my head around it.
In addition, since you're going to be mutating the string, why not start with an `NSMutableString` to avoid potentially allocating `n^2` amount of memory?

File: Calculator/Podfile.lock [hunk 3]
```diff
@@ -22,4 +22,6 @@ SPEC CHECKSUMS:
   OCMock: 18c9b7e67d4c2770e95bb77a9cc1ae0c91fe3835
   OHHTTPStubs: f7d1604d04d37d055460c0ef9af01bb249ded0fa
 
-COCOAPODS: 0.36.4
+PODFILE CHECKSUM: e4ea1d31e1c4e8dd512d26af4ad712b8d1e86ccc
+
+COCOAPODS: 1.0.1
```
Comment (nielubowicz) [inline Calculator/Podfile.lock]: I would prefer that you _not_ update the pods library and references in your PR. There are a number of changes included in this PR that are not at all related to the small method and unit test found herein. Please use the already-existing version of Cocoapods, unless otherwise specified or approved by @eccarmichael.

File: Calculator/CalculatorTests/ModelsTests/MCACalculatorTests.m [hunk 4]
```diff
@@ -148,4 +148,19 @@ - (void)testMCACalculator_pushOperatorWithOperand_shouldNotSetExpressionComplete
     //Assert
     XCTAssertFalse(expressionComplete, @"Expected push operand with operator to not set expression complete.");
 }
+
```
Comment (nielubowicz) [inline Calculator/CalculatorTests/ModelsTests/MCACalculatorTests.m]: Please add several more unit tests to cover the full breadth of the binary conversion operation: as I mentioned in one of the unit testing intro videos, it is best practice to cover _all_ cases, not just a "normal" use case.

I like to add at least the following types of tests:
- Good data (this one already exists  🎉 !)
- Nil data
- "Bad" data --- Does your binary conversion support negative numbers? If so, add one in the "Good" section, if not, add one in the "bad" section. i.e. in the first case, `@"-1"` should yield a valid string, but in the second case, the function should either throw or return `nil` (both of which can be addressed in the `// Assert ` section of your tests)
- "Garbage" data -- I know the calculator as it stands does not accept string input, but I'd like to see tests against it anyway: try to convert "Chris Is The Greatest" to a binary number ;)

I implemented the untested items, locally, and all of the following tests fail : "Bad", nil, and Garbage.

File: Calculator/Podfile.lock (hunk 3 above)
Comment (mob-jen) [inline Calculator/Podfile.lock]: I'm not quite sure where that came from. I will revert that.

**Comment by mob-jen** [comment]
Closing. I'm going to approach this differently.
